wormchain/x/wormhole/client/cli/genesis.go

```diff
@@ -405,6 +412,115 @@ func CmdGenerateGuardianSetUpdatea() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "generate-slashing-params-update-vaa",
+		Short: "generate and sign a governance vaa to update the slashing params",
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+
+			privateKeys := []*ecdsa.PrivateKey{}
+			privateKeysFiles, err := cmd.Flags().GetStringArray(FLAG_KEY)
+			if err != nil {
+				return err
+			}
+			for _, privFile := range privateKeysFiles {
+				priv, err := ImportKeyFromFile(privFile)
+				if err != nil {
+					return err
+				}
+				privateKeys = append(privateKeys, priv)
+			}
+			v, err := parseVaaFromFlags(cmd)
+			if err != nil {
+				return err
+			}
+
+			signedBlocksWindow, err := cmd.Flags().GetUint64(FLAG_SIGNED_BLOCKS_WINDOW)
+			if err != nil {
+				return err
+			}
+
+			minSignedPerWindow, err := cmd.Flags().GetString(FLAG_MIN_SIGNED_PER_WINDOW)
+			if err != nil {
+				return err
+			}
+			minSignedPerWindowDec, err := sdk.NewDecFromStr(minSignedPerWindow)
+			if err != nil {
+				return err
+			}
+			if minSignedPerWindowDec.LT(sdk.ZeroDec()) || minSignedPerWindowDec.GT(sdk.OneDec()) {
+				return fmt.Errorf("min signed per window must be greater than or equal to 0.000000000000000000 and less than or equal to 1.000000000000000000")
+			}
+
+			downtimeJailDuration, err := cmd.Flags().GetUint64(FLAG_DOWNTIME_JAIL_DURATION)
+			if err != nil {
+				return err
+			}
+
+			slashFractionDoubleSign, err := cmd.Flags().GetString(FLAG_SLASH_FRACTION_DOUBLE_SIGN)
+			if err != nil {
+				return err
+			}
+			slashFractionDoubleSignDec, err := sdk.NewDecFromStr(slashFractionDoubleSign)
+			if err != nil {
+				return err
+			}
+			if slashFractionDoubleSignDec.LT(sdk.ZeroDec()) || slashFractionDoubleSignDec.GT(sdk.OneDec()) {
+				return fmt.Errorf("slash fraction double sign must be greater than or equal to 0.000000000000000000 and less than or equal to 1.000000000000000000")
+			}
+
+			slashFractionDowntime, err := cmd.Flags().GetString(FLAG_SLASH_FRACTION_DOWNTIME)
+			if err != nil {
+				return err
+			}
+			slashFractionDowntimeDec, err := sdk.NewDecFromStr(slashFractionDowntime)
+			if err != nil {
+				return err
+			}
+			if slashFractionDowntimeDec.LT(sdk.ZeroDec()) || slashFractionDowntimeDec.GT(sdk.OneDec()) {
+				return fmt.Errorf("slash fraction downtime must be greater than or equal to 0.000000000000000000 and less than or equal to 1.000000000000000000")
+			}
+
+			slashingUpdate := make([]byte, 40)
+			binary.BigEndian.PutUint64(slashingUpdate[:8], signedBlocksWindow)
+			binary.BigEndian.PutUint64(slashingUpdate[8:16], minSignedPerWindowDec.BigInt().Uint64())
+			binary.BigEndian.PutUint64(slashingUpdate[16:24], downtimeJailDuration*uint64(time.Second))
+			binary.BigEndian.PutUint64(slashingUpdate[24:32], slashFractionDoubleSignDec.BigInt().Uint64())
+			binary.BigEndian.PutUint64(slashingUpdate[32:40], slashFractionDowntimeDec.BigInt().Uint64())
+
+			action := vaa.ActionSlashingParamsUpdate
+			chain := 3104
```

Comment:
can we reference wormchain by name instead of by number here so that things are a bit more readable?